src/vm/functions/mod.rs

```diff
@@ -397,3 +408,113 @@ fn special_contract_of(args: &[SymbolicExpression], env: &mut Environment, conte
     let contract_principal = Value::Principal(PrincipalData::Contract(contract_identifier.clone()));
     Ok(contract_principal)
 }
+
+fn special_principal_of(args: &[SymbolicExpression], env: &mut Environment, context: &LocalContext) -> Result<Value> {
+    // (principal-of (..))
+    // arg0 => (buff 33)
+    check_argument_count(1, args)?;
+
+    runtime_cost!(cost_functions::PRINCIPAL_OF, env, 0)?;
+
+    let param0 = eval(&args[0], env, context)?;
+    let pub_key = match param0 {
+        Value::Sequence(SequenceData::Buffer(BuffData { ref data })) => {
+            if data.len() != 33 {
+                return Err(CheckErrors::TypeValueError(BUFF_33, param0).into())
+            }
+            data
+        },
+        _ => return Err(CheckErrors::TypeValueError(BUFF_33, param0).into())
+    };
+
+    let addr = StacksAddress::from_public_keys(C32_ADDRESS_VERSION_TESTNET_SINGLESIG, &AddressHashMode::SerializeP2PKH, 1, &vec![Secp256k1PublicKey::from_slice(&pub_key).unwrap()]).unwrap();
+    let principal = addr.to_account_principal();
+
+    Ok(Value::Principal(principal))
+}
+
+fn special_secp256k1_recover(args: &[SymbolicExpression], env: &mut Environment, context: &LocalContext) -> Result<Value> {
```

Comment:
I think we should put these functions in a module rather than adding them here (perhaps `arithmetic`?)

Reply:
I followed the lead of how/where `sha...` and `keccak...` were done.  Not sure, these would belong in `arithmetic` but maybe all these could be moved into a new module named `crypto` maybe?

Reply:
Yeah, a new module `crypto` makes sense

Reply:
see 3907ed5